app/javascript/components/upload/ClusteringFileForm.jsx

```diff
@@ -20,20 +23,63 @@ export default function ClusteringFileForm({
   associatedClusterFileOptions=[],
   updateCorrespondingClusters,
   bucketName,
-  isInitiallyExpanded
+  isInitiallyExpanded,
+  isAnnDataExperience
 }) {
   const spatialClusterAssocs = file.spatial_cluster_associations
     .map(id => associatedClusterFileOptions.find(opt => opt.value === id))
   const validationMessages = validateFile({
     file, allFiles, allowedFileExts, requiredFields
   })
 
+
+  /** create the tooltip and message for the .obsm key names section */
+  function obsmKeyNameMessage() {
```

Comment:
Trivial blocker: React component names start with a capital letter per SCP convention.

```suggestion
  function ObsmKeyNameMessage() {
```